### PR TITLE
[Fix] Fix the sortorder For Dance results score category in web project to match collated.

### PR DESCRIPTION
**Pull request by davidsmonkeys**
## Description

Order the scores within the dance list response so that they are congruent with the collated sores column order.

It addresses the problem raised in the DertInfo-web project here: https://github.com/dertinfo/dertinfo-web/issues/21

Fixes #dertinfo-web 21 (issue)

The web project should likely deal with this issue directly but this is the shortest path to a functioning solution for dert2025 . 

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

We have validated that this change does address the sort order problem and appears to only affect this API call so isexpected to be isolated. 

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules 
### :bulb: PR Summary generated by FirstMate

**Overview**: Fixed the sort order for Dance results by implementing a method to order scores.

**Changes**:
*Score Ordering Logic:*
- Added `OrderScoresByScoreCategory` method in `Dance.cs` to sort scores by `ScoreCategory.SortOrder`.
- Updated `ListForCompetition` method in `DanceService.cs` to call the new sorting method for each dance.

**Documentation:**
- Added comments in `ListForCompetition` to clarify the purpose of the sorting logic.

**TLDR**: Implemented score ordering for Dance results to match collated results; focus on the new sorting method in `Dance.cs` and its integration in `DanceService.cs`.

<sub><sup>Generated by FirstMate and automatically updated on every commit.</sup></sub>

### CODE REVIEW

**Review by dertinfo — APPROVED** [review]
It's note great as not very clean or efficient but I'll take it.

**Comment by davidsmonkeys** [comment]
This PR does successfully ensure that the order or the dance scores on the score categories is correct however in testing we identified that this is not also the case on collated. and that the collated scores are in the correct order by chance. Likely due to the way that the compeientation is originally setup by the system and the order it creates the score categories in the db.
